packages/compass-shell/src/modules/worker-runtime/worker.js

```diff
@@ -0,0 +1,64 @@
+import { parentPort } from 'worker_threads';
+import { ElectronRuntime } from '@mongosh/browser-runtime-electron';
+import { CliServiceProvider } from '@mongosh/service-provider-server';
+import { exposeAll, createCaller } from './rpc';
+import { serializeResult, serializeError } from './worker-shell-result';
+
+let provider = null;
+let runtime = null;
+
+const evaluationListener = createCaller(
+  ['onPrint', 'onPrompt', 'toggleTelemetry'],
+  parentPort
+);
+
+async function init(uri, options, cliOptions) {
+  provider = await CliServiceProvider.connect(uri, options, cliOptions);
+  runtime = new ElectronRuntime(provider /** , TODO: messageBus? */);
+
+  runtime.setEvaluationListener(evaluationListener);
+
+  return true;
+}
+
+async function evaluate(code) {
+  try {
+    if (!runtime) {
+      throw new Error('Runtime is not initiated yet');
+    }
+
+    const result = await runtime.evaluate(code);
```

Comment:
I think in order for the `SIGINT` to work reliably, we’d also need a dummy `process.on('SIGINT', noop)` listener in the main thread of the child process? Otherwise a mis-timed Ctrl+C would still bring the child process down, right?

Reply:
Yep, totally something to handle better than it is done here 😄